test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,360 @@
+            stop_resp = self.stop_vm(vm_id)
+            update_resp = self.update_vm(vm_id, extraconfig)
+            start_resp = self.start_vm(vm_id)
+
+            host_id = start_resp.hostid
+            host = list_hosts(
+                self.apiclient,
+                id=host_id,
+                hypervisor='kvm')
+
+            instance_name = response.instancename
+            host_ipaddress = host[0].ipaddress
+
+            ssh_client = SshClient(host_ipaddress, port=22,
+                                   user=self.hostConfig['username'],
+                                   passwd=self.hostConfig['password'])
+            virsh_cmd = 'virsh dumpxml %s' % instance_name
+            xml_res = ssh_client.execute(virsh_cmd)
+            xml_as_str = ''.join(xml_res)
+
+            extraconfig_decoded_xml = '<config>' + urllib.unquote(extraconfig) + '</config>'
+
+            # Root XML Elements
+            parser = etree.XMLParser(remove_blank_text=True)
+            domain_xml_root = ET.fromstring(xml_as_str, parser=parser)
+            decoded_xml_root = ET.fromstring(extraconfig_decoded_xml, parser=parser)
+            for child in decoded_xml_root:
+                find_element_in_domain_xml = domain_xml_root.find(child.tag)
+
+                # Fail if extra config is not found in domain xml
+                self.assertNotEquals(
+                    0,
+                    len(find_element_in_domain_xml),
+                    'Element tag from extra config not added to VM'
+                )
+
+                # Compare found XML node with extra config node
+                is_a_match = self.elements_equal(child, find_element_in_domain_xml)
+                self.assertEquals(
+                    True,
+                    is_a_match,
+                    'The element from tags from extra config do not match with those found in domain xml'
+                )
+        finally:
+            self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_vmware(self):
+        if self.hypervisor.lower() != 'vmware':
+            raise self.skipTest("Skipping test case for non-vmware hypervisor")
+
+        extraconfig = None
+        response = self.deploy_vm('vmware', extraconfig)
+        host_id = response.hostid
+        host = list_hosts(
+            self.apiclient,
+            id=host_id)
+
+        instance_name = response.instancename
+        host_ipaddress = host[0].ipaddress
+
+        ssh_client = SshClient(host_ipaddress, port=22,
+                               user=self.hostConfig['username'],
+                               passwd=self.hostConfig['password'])
+
+        extraconfig_decoded = urllib.unquote(extraconfig)
+        config_arr = extraconfig_decoded.splitlines()
+
+        for config in config_arr:
+            grep_config = "cat /vmfs/volumes/eaa90eaf-b789d201/%s/%s.vmx | grep -w '%s'" % (
```

Comment:
hard-coded?

Reply:
you can try using the vmware client on the host to get the info about the vmx file: 
[root@trl-1748-k-M7-bstoyanov-esxi1:~] esxcli vm process list
```
i-2-26-VM
   World ID: 77590
   Process ID: 0
   VMX Cartel ID: 77589
   UUID: 42 06 ce 56 19 ff dd 22-7d a9 00 8c 79 85 e0 09
   Display Name: i-2-26-VM
   Config File: /vmfs/volumes/dc7ba692-c6334326/i-2-26-VM/i-2-26-VM.vmx
```

Reply:
fixed